### ISSUE TITLE
Clear Terminal only when changing directory

### ISSUE THREAD

**Issue by theRoboxx** [leo-arch/CliFM#292]
**Is your feature request related to a problem? Please describe.**
Sometimes there is a lot of output and overflow from an application output text that I want to read. But currently, the text will me cleared, if it overflows the window. Additionally I want to clear the screen automatic when switching the directory for better visibility.

**Describe the solution you'd like**
There should be a new option to only clear the terminal window when switching the directory and NOT when the output of an application is too much to display in the current window.
tl;dr: Only clear terminal when switching directory and never otherwise.

**Comment by leo-arch** [comment]
> the text will me cleared, if it overflows the window

This is not intended behavior.  When `ClearScreen` is set to true (default) the screen is cleared and the current list of files reprinted **whenever some file has been added and/or removed/renamed**, but not when the output of a command overflows the window. However, it might happen that sometimes your shell command does modify the files list (in which case the list of files is reprinted) **and** you still need to read the output (which at this point is lost).

Could you provide an example command?

**Comment by leo-arch** [comment]
With the latest commit, you can set `ClearScreen` to `internal` in the config file to prevent external commands, even if they create/remove/rename a file in the current directory, from refreshing the current list of files. For example:

```
echo "test" > myfile
```
**Will not** refresh the screen (since `echo` is an _external_ command). But,

```
n myfile
```
will, because `n` is an _internal_ command.

**Comment by theRoboxx** [comment]
Thanks, this behavior is much better but for my use case there is something missing.

Would it be possible to clear and refresh the screen only if switching the directory but if you add/remove files in the current directory and there is already some written output, the list of the current files/folders is refreshed AND the old terminal output is still present and not deleted?

For example if I run:
```
echo "test" > myfile
```
and
```
r myfile
```
I would like to get a refresh of the file list with myfile missing but still seeing the written command 
```
<0> $ echo "test" > myfile
<0> r myfile
```
 in the terminal.

**Comment by theRoboxx** [comment]
Maybe this is a more precise explanation:
It would be like setting ClearScreen=false and AutoLs=true. but with the addition that the file/folder list gets refreshed on top and not in a new line and the commands from the past are still visible while in the directory.

**Comment by leo-arch** [comment]
> I would like to get a refresh of the file list with myfile missing but still seeing the written command

This would be easier if we were using a curses environment, where you have multiple independent screens or panes. But don't use curses, and this means: we have only one screen. If we reprint the files list, previous output is lost.

However, we do keep a commands history, so that it is theoretically possible to keep those commands executed in the current directory in sight (though this does not include shell commands output) even if the files list is reprinted.

What we need is a good way (and a simple logic) to know what commands to keep in sight.

**Comment by leo-arch** [comment]
This is what I came up with: by setting `PrintDirCmds` to `true` in the config file, commands executed in the current directory will be printed before the prompt, but only while you are in the current directory: change the directory and this commands list will be lost.

Give it a shot and let me know what you think.

**Comment by theRoboxx** [comment]
Works great :100:  .
I will use it a bit and give feedback if somethings breaks.

Thanks again for implementing this feature ^_^.
